From 0df71d83257ef15063ec4dad93349466fc09af3d Mon Sep 17 00:00:00 2001
From: Heinrich Schuchardt <xypron.glpk@gmx.de>
Date: Thu, 20 Aug 2020 19:53:41 +0200
Subject: [PATCH 1/1] lib/hashtable: remove superfluous check

We assign first_deleted = 0. There is no need to check its value without
any further assignment in between.

Signed-off-by: Heinrich Schuchardt <xypron.glpk@gmx.de>
---
indicated by cppcheck
---
 lib/hashtable.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/lib/hashtable.c b/lib/hashtable.c
index 4a8c50b4b8..7c08f5c805 100644
--- a/lib/hashtable.c
+++ b/lib/hashtable.c
@@ -324,8 +324,7 @@ int hsearch_r(struct env_entry item, enum env_action action,
 		 */
 		unsigned hval2;
 
-		if (htab->table[idx].used == USED_DELETED
-		    && !first_deleted)
+		if (htab->table[idx].used == USED_DELETED)
 			first_deleted = idx;
 
 		ret = _compare_and_overwrite_entry(item, action, retval, htab,
-- 
2.28.0

